db/notifications.go

```diff
@@ -111,3 +142,55 @@ func NotificationSendSummary(ctx context.Context, id string, window time.Duratio
 	err := ctx.DB().Raw(query, id, window).Row().Scan(&earliest, &count)
 	return earliest.Time, count, err
 }
+
+func GetMatchingNotificationSilencesCount(ctx context.Context, resources models.NotificationSilenceResource) (int64, error) {
+	_ = ctx.DB().Use(extraClausePlugin.New())
+
+	query := ctx.DB().Debug().Model(&models.NotificationSilence{})
+
+	// Initialize with a false condition,
+	// if no resources are provided, the query won't return all records
+	orClauses := ctx.DB().Where("1 = 0")
+
+	if resources.ConfigID != nil {
+		orClauses = orClauses.Or("config_id = ?", *resources.ConfigID)
+
+		// recursive stuff
```

Comment:
What does this query look like when generated ?

Reply:
```sql
With path_cte AS (
  SELECT path FROM config_items Where id = $1
)
SELECT COUNT(*)
  FROM notification_silences
  CROSS JOIN path_cte
WHERE (config_id = $1 OR (recursive = true AND path_cte LIKE '%s' || config.id::TEXT || '%s'))
  AND from <= NOW()
  AND until >= NOW()
```

Reply:
Do we have a test for this, because I think the path query is wrong - If there is an existing silence on a config_id (parent), then we also silence with config.path like parent.path + "%'

Reply:
> If there is an existing silence on a config_id (parent), then we also silence with config.path like parent.path + "%'

This would only work for root components no? So if you have silenced on a deployment , that would not reciprocate to the pod because the pod's path is `<namespace>.<deployment>.<pod>`. It would work if you've silenced the namespace however.

We do have a test for this. https://github.com/flanksource/mission-control/pull/1364/files#diff-c18508950d90e5dd1426ae0b6a32c3f68477ce565551303c2835a701ffc16b9eR87-R98

A parent is silenced and consequently its child and grandchild is also silenced.

Reply:
The pods path should start with the namespace/deployment path